extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/QuarkusJaegerMetricsFactory.java

```diff
@@ -21,6 +21,9 @@
 
     @Override
     public Counter createCounter(final String name, final Map<String, String> tags) {
+        // On restart (e.g. in dev mode) Jaeger registers
+        // the same metrics which would result in exception
+        registry.remove(name);
```

Comment:
On Jaeger creates metrics. On runtime restart a new tracer is created therefore a set of metrics is registered. On multiple restarts the same set of metrics are being registered which causes 

```
Caused by: java.lang.IllegalArgumentException: A metric with metricID MetricID{name='jaeger_tracer_reporter_queue_length', tags=[]} already exists
	at io.smallrye.metrics.MetricsRegistryImpl.register(MetricsRegistryImpl.java:129)
	at io.quarkus.jaeger.runtime.QuarkusJaegerMetricsFactory.createGauge(QuarkusJaegerMetricsFactory.java:48)
	at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:66)
	... 37 more
```

Does anybody know a better solution to this?

Reply:
@jmartisk perhaps?

Reply:
@jmartisk Any idea about this?

Reply:
Oh, my deepest apologies for such late reply! The notifications got buried somewhere and I probably skipped over them when there we too many of them to have a thorough look.
I suppose you should use a shutdown task for any metric that the extension creates so that it will get removed on each live reload 
See https://github.com/quarkusio/quarkus/blob/master/extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java#L110
I am seriously considering changing the Metrics extension so that all metrics will be removed automatically though, so this will not be needed anymore

Reply:
No problem @jmartisk !

Reply:
Or is this something that happens dynamically at runtime, rather than through a recorder? In that case I suppose you can't use a shutdown task, and I'd say the right thing to do is to remove them like you do atm, until we performed the Metrics change I mentioned.

Reply:
thanks @jmartisk then I will keep this here.